File: backend/blog-service/src/main/java/com/lily/blogservice/dto/BlogDocument.java
package com.lily.blogservice.dto;

import java.time.LocalDateTime;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
@EqualsAndHashCode
public class BlogDocument {

  private Long favoriteId;

  private String blogName;

  private String title;

  private String contents;

  private String url;

  private String thumbnail;

  private LocalDateTime datetime;

  public boolean getLike() {
    return favoriteId != null;
  }

  public int getUrlHashCode() {
    return url.hashCode();
  }

}

File: backend/blog-service/src/main/kotlin/com/lily/blogservice/data/BlogDocument.kt
package com.lily.blogservice.data

import java.time.LocalDateTime

class BlogDocument(
    val favoriteId:Long?,
    val blogName:String="",
    val title:String="",
    val contents:String="",
    val url:String="",
    val thumbnail:String="",
    val datetime: LocalDateTime,
    ) {

    fun getUrlHashCode():Int = url.hashCode()
}